docs(readme): comply new policy template INTER-295

Orkuncakilkaya opened the pull request.

Orkuncakilkaya committed:
docs(readme): comply new policy template

Orkuncakilkaya committed:
docs(readme): add issues link

Orkuncakilkaya committed:
docs(readme): add node version 18

necipallef commented on README.md:
```diff
@@ -23,11 +23,24 @@
 
+## Requirements
+
+This library needs these to be in order to build and run:
```
We need to mention here that the integration backwards-compatibility is provided by Cloudflare, and there are only requirements for local development. We can mention [this field](https://github.com/fingerprintjs/fingerprintjs-pro-cloudflare-worker/blob/rc/wrangler.toml#L6) as well.

Orkuncakilkaya replied:
Updated with even better, as Juraj Suggested I extended requirements section with two subsections called "Usage Requirements" and "Development Requirements".

necipallef commented on README.md:
```diff
@@ -23,11 +23,24 @@
+
+This library needs these to be in order to build and run:
+
+- Node 16 or Node 18
```
I think we should mention Node20 and can even remove Node 16

Orkuncakilkaya replied:
Done 👍

necipallef commented on README.md:
```diff
@@ -23,11 +23,24 @@
 
+## Support
+
+To report problems, ask questions or provide feedback, please use [Issues](https://github.com/fingerprintjs/fingerprintjs-pro-cloudflare-worker/issues). If you need private support, you can email us at [oss-support@fingerprint.com](mailto:oss-support@fingerprint.com).
```
Is `oss-support@fingerprint.com` the correct email address here?

Orkuncakilkaya replied:
Yes, according the guideline "Writing DX GitHub Readmes" provided by Juraj at our Notion documents.

necipallef commented on README.md:
```diff
@@ -23,11 +23,24 @@
+- Node 16 or Node 18
+- Typescript 4
+- Playwright (with a headless browser preferably chromium)
+- [Wrangler](https://developers.cloudflare.com/workers/wrangler/install-and-update/)
```
There are different `wrangler` versions out there, `v1`, `v2` and `v3`. In my case, I started using `v1` and then switched to `v2`, and never tried `v3`. We probably support `v2` and `v3` and not `v1` but need to make sure and mention it here.

Orkuncakilkaya replied:
We support v3+ I mentioned in the readme

necipallef reviewed (changes requested):
left comments

Orkuncakilkaya committed:
docs(readme): wrangler version

Orkuncakilkaya committed:
docs(readme): wrangler version

Orkuncakilkaya committed:
docs(readme): update requirements

change node version to 20
update typescript version to 4 and plus
introduce usage and development reqs

JuroUhlar commented on README.md:
```diff
@@ -23,11 +23,30 @@
+- Playwright
+- [Wrangler](https://developers.cloudflare.com/workers/wrangler/install-and-update/) v3+
+
 ## Setup process
```
* This seems outdated, you don't need to contact support to setup the Cloudflare integration. 
* The Readme should point people to https://dev.fingerprint.com/docs/cloudflare-integration follow the set up guide using the installation wizard.

Orkuncakilkaya replied:
Updated here

JuroUhlar commented on README.md:
```diff
@@ -23,11 +23,30 @@
+
+### Usage Requirements
+
+Please see the [Setup Process](#setup-process) section
```
The integration actually has a well defined usage requirements defined https://dev.fingerprint.com/docs/cloudflare-integration#setup 

You could briefly list them here and point to the guide for more details

Orkuncakilkaya replied:
Updated usage requirements

JuroUhlar commented on README.md:
```diff
@@ -23,11 +23,30 @@
+
+Please see the [Setup Process](#setup-process) section
+
+### Development Requirements
```
The information hierarchy is off here, Setup/Getting started is universal and much more important than Development requirements. Consider moving them below or to `contributing.md`

Orkuncakilkaya replied:
Moved development requirements inside contributing.md file

JuroUhlar commented on README.md:
```diff
@@ -23,11 +23,30 @@
 # Fingerprint Pro Cloudflare worker
 The Cloudflare worker is responsible for delivering the latest fingerprinting client-side logic and proxying identification requests and responses between your site and Fingerprint Pro's APIs.
```
Consider aligning the introduction and setup sections with Cloudfront: https://github.com/fingerprintjs/fingerprint-pro-cloudfront-integration 
and Azure https://github.com/fingerprintjs/fingerprint-pro-azure-integration

Orkuncakilkaya replied:
Updated description

Orkuncakilkaya committed:
docs(readme): setup process without contacting support

Orkuncakilkaya committed:
docs(readme): usage requirements

Orkuncakilkaya committed:
docs(readme): move development reqs to contributing

Orkuncakilkaya committed:
docs(readme): update description

JuroUhlar commented on README.md:
```diff
@@ -21,13 +21,36 @@
+- Proxying download requests of the latest Fingerprint Pro JS Agent between your site and Fingerprint CDN.
+- Proxying identification requests and responses between your site and Fingerprint Pro's APIs.
+
+This [improves](https://dev.fingerprint.com/docs/cloudfront-proxy-integration#the-benefits-of-using-the-cloudfront-integration) both accuracy and reliability of visitor identification and bot detection on your site.
```
```suggestion
This [improves](https://dev.fingerprint.com/docs/cloudflare-integration#the-benefits-of-using-the-cloudflare-integration) both accuracy and reliability of visitor identification and bot detection on your site.
```

JuroUhlar commented on README.md:
```diff
@@ -21,13 +21,36 @@
+  - [Cloudflare API Token](https://dev.fingerprint.com/docs/cloudflare-integration#cloudflare-api-token)
+- FingerprintPro Account
+
+By just following the wizard with your credentials it's easy to setup. Please see the [Setup Process](#setup-process) section
```
@Orkuncakilkaya These are not the requirements listed in the docs. Getting Cloudflare account ID and token is covered in the guide, therefore these are not requirements. Requirements or pre-requisites are "stuff the guide/tool assumes you already have before you start", not "stuff the guide helps you set up".

```suggestion
* [Fingerprint Pro account](https://dashboard.fingerprint.com/signup) with the _Owner_ role assigned.
* A website served by Cloudflare. For maximum accuracy benefits, your website should be [proxied by Cloudflare](https://developers.cloudflare.com/dns/manage-dns-records/reference/proxied-dns-records/) (not DNS-only). 

See the [Cloudflare integration guide](https://dev.fingerprint.com/docs/cloudflare-integration#setup) for more details. 

```

JuroUhlar commented on README.md:
```diff
@@ -21,13 +21,36 @@
+
+By just following the wizard with your credentials it's easy to setup. Please see the [Setup Process](#setup-process) section
 
 ## Setup process
```
```suggestion
## How to install
```

JuroUhlar commented on README.md:
```diff
@@ -21,13 +21,36 @@
-The process consists of two steps. One needs to set up worker creation together with our support at [support@fingerprint.com](mailto:support@fingerprint.com). Afterwards, the Fingerprint Pro JS agent on the site needs to be configured to communicate with the worker.
 
+Setup process is simple enough. You just follow a setup wizard and it's ready to use.
 You can find the full documentation at [https://dev.fingerprint.com/docs/cloudflare-integration](https://dev.fingerprint.com/docs/cloudflare-integration).
```
Tech writing tips: 
* Don't make qualitative statements like "this is simple", that's highly subjective
* Links should have meaningful text, not URLs 


```suggestion
You can install the Cloudflare integration using an [installation wizard](https://dashboard.fingerprint.com/integrations) in the Fingerprint dashboard. 

See the [Cloudflare integration guide](https://dev.fingerprint.com/docs/cloudflare-integration#setup) for more details.
```

Orkuncakilkaya committed:
docs(readme): update incorrect link

Co-authored-by: Juraj Uhlar <juro.uhlar@gmail.com>

Orkuncakilkaya committed:
docs(readme): update usage requirements

Co-authored-by: Juraj Uhlar <juro.uhlar@gmail.com>

Orkuncakilkaya committed:
docs(readme): update setup process

Co-authored-by: Juraj Uhlar <juro.uhlar@gmail.com>

Orkuncakilkaya committed:
docs(readme): rename setup process to how to install

Co-authored-by: Juraj Uhlar <juro.uhlar@gmail.com>
